tools/src/tester/helpers.ts

```diff
@@ -1,8 +1,152 @@
-import { type Evaluation, Result } from './types/eval.types'
+import { type Evaluation, Result, EvaluationWithOutput, StoryOutputs, StoryEvaluation, OutputReference } from './types/eval.types'
+import { ActualResponse, ChapterRequest, Output, Parameter, Payload, Story } from './types/story.types'
+import _ from 'lodash'
 
-export function overall_result (evaluations: Evaluation[]): Result {
+export function overall_result(evaluations: Evaluation[]): Result {
   if (evaluations.some(e => e.result === Result.ERROR)) return Result.ERROR
   if (evaluations.some(e => e.result === Result.FAILED)) return Result.FAILED
   if (evaluations.some(e => e.result === Result.SKIPPED)) return Result.SKIPPED
   return Result.PASSED
 }
+
+export function extract_output_values(response: ActualResponse, chapterOutput?: Output): EvaluationWithOutput {
```

Comment:
Refactor into `ChapterOutput`, I think it should be constructed from `ActualResponse`, be a method on `ActualResponse`, or something like that.